R/plot_train_valid_error.R

```diff
@@ -0,0 +1,167 @@
+      y_valid_pred <- class::knn(X_train, X_valid, y_train, k = param_vec[i])
+
+    } else if (model_name == 'decision tree'){
+
+      if (param_name != 'maxdepth'){
+        stop("Sorry, only the hyperparameter 'maxdepth' is allowed for a 'decision tree' model.")
+      }
+      classifier <- rpart::rpart(formula = Target ~ .,
+                                 data = data,
+                                 control = list(maxdepth = param_vec[i]))
+      y_train_pred <- stats::predict(classifier, newdata = X_train, type = 'class')
+      y_valid_pred <- stats::predict(classifier, newdata = X_valid, type = 'class')
+
+    } else if (model_name == 'svm') {
+
+      if (param_name == 'cost') {
+        classifier <- e1071::svm(formula = Target ~.,
+                                 data = data,
+                                 type = 'C-classification',
+                                 kernel = 'linear',
+                                 cost = param_vec[i])
+
+      } else if (param_name == 'gamma') {
+        classifier <- e1071::svm(formula = Target ~.,
+                                 data = data,
+                                 type = 'C-classification',
+                                 kernel = 'linear',
+                                 gamma = param_vec[i])
+
+      } else {
+        stop("Sorry, only the hyperparameters, 'cost' and 'gamma', are allowed for a 'svm' model.")
+      }
+
+      y_train_pred <- stats::predict(classifier, newdata = X_train, type = 'class')
+      y_valid_pred <- stats::predict(classifier, newdata = X_valid, type = 'class')
+
+    } else if (model_name == 'random forests') {
+
+      if (param_name != 'ntree') {
+        stop("Sorry, only the hyperparameter 'ntree' is allowed for a 'random forests' model.")
+      }
+
+      classifier <- randomForest::randomForest(x = X_train,
+                                               y = y_train,
+                                               ntree = param_vec[i])
+      y_train_pred <- stats::predict(classifier, newdata = X_train, type = 'class')
+      y_valid_pred <- stats::predict(classifier, newdata = X_valid, type = 'class')
+
+    } else {
+
+      stop("Sorry, the model_name should be chosen from 'knn', 'decision tree', 'svc' and 'random forests'.")
+    }
+
+    df$train[i] = sum(y_train_pred != y_train)/length(y_train)
+    df$valid[i] = sum(y_valid_pred != y_valid)/length(y_valid)
+  }
+
+
+  plot <- tidyr::gather(df, "dataset", "error", - para) %>%
+    ggplot2::ggplot(ggplot2::aes(para, error, color = dataset)) +
+    ggplot2::geom_line() +
+    ggplot2::labs(x = param_name,
+                  y = "Error",
+                  title = paste(model_name, "train and valid errors vs.", param_name)) +
+    ggplot2::theme(plot.title = ggplot2::element_text(size=16),
+                   axis.title = ggplot2::element_text(size=14),
+                   text = ggplot2::element_text(size=13))
+  plot
```

Comment:
Thanks for pointing out that. I'll make the following changes to avoid using `%>%` as well.
```suggestion
  df <- tidyr::gather(df, "dataset", "error", - para)
  ggplot2::ggplot(df, ggplot2::aes(para, error, color = dataset)) +
    ggplot2::geom_line() +
    ggplot2::labs(x = param_name,
                  y = "Error",
                  title = paste(model_name, "train and valid errors vs.", param_name)) +
    ggplot2::theme(plot.title = ggplot2::element_text(size=16),
                   axis.title = ggplot2::element_text(size=14),
                   text = ggplot2::element_text(size=13))
```